Orm/Xtensive.Orm/Orm/Linq/TranslatorContext.cs

```diff
@@ -84,6 +87,17 @@ public ApplyParameter GetApplyParameter(CompilableProvider provider)
       return parameter;
     }
 
+    public string[] GetAllTags()
+    {
+      if (Domain.Configuration.TagsLocation == TagsLocation.Nowhere)
+        return Array.Empty<string>();
+
+      var tags = applyParameters.Keys.OfType<TagProvider>().Select(p => p.Tag).ToList();
+      if (tags.Count == 0)
+        return Array.Empty<string>();
+      return tags.ToArray();
```

Comment:
Can we replace method's return type with `IReadOnlyList<string>` so we don't have to copy content of newly created list to array?
So the implementation could be
```csharp
    public IReadOnlyList<string> GetAllTags()
    {
      if (Domain.Configuration.TagsLocation == TagsLocation.Nowhere)
        return Array.Empty<string>();

      return applyParameters.Keys.OfType<TagProvider>().Select(p => p.Tag).ToList();
    }
```

Reply:
Yes, we can since it is internal class. For public APIs I would prefer having real read-only collections, though.

Reply:
Arrays are not read-only btw as you can replace any item there.